### ISSUE TITLE
Crash "try? Realm()" v4.4.1

### ISSUE THREAD

**Issue by Algashev** [realm/realm-cocoa#6578]
## Goals
Perform the migration

## Expected Results
Migration is performed. Apps not crashing

## Actual Results
On some cases apps crashes
```
Crashed: com.apple.main-thread
0  libsystem_kernel.dylib         0x1ae4fed88 __pthread_kill + 8
1  libsystem_pthread.dylib        0x1ae41b74c pthread_kill$VARIANT$armv81 + 216
2  libsystem_c.dylib              0x1ae36a644 abort + 100
3  Realm                          0x10420bee0 please_report_this_error_to_help_at_realm_dot_io + 10
4  Realm                          0x10420c1ac realm::util::terminate_internal(std::__1::basic_stringstream<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) + 264
5  Realm                          0x10420c320 realm::util::terminate(char const*, char const*, long, std::initializer_list<realm::util::Printable>&&) + 324
6  Realm                          0x103f269cc void realm::util::terminate<unsigned long, int>(char const*, char const*, long, unsigned long, int) + 54
7  Realm                          0x1041b9270 realm::Table::init(unsigned long, realm::ArrayParent*, unsigned long, bool) + 654
8  Realm                          0x1040ebcd4 realm::Group::create_table_accessor(unsigned long) + 240
9  Realm                          0x1040ebb64 realm::Group::do_get_table(unsigned long, bool (*)(realm::Spec const&)) + 260
10 Realm                          0x103f1f530 realm::ObjectStore::get_schema_version(realm::Group const&) + 52
11 Realm                          0x10400b80c realm::Realm::read_schema_from_group_if_needed() + 68
12 Realm                          0x10400a7e4 realm::Realm::read_group() + 44
13 Realm                          0x104009a5c realm::Realm::Realm(realm::Realm::Config, std::__1::shared_ptr<realm::_impl::RealmCoordinator>) + 368
14 Realm                          0x103f3d608 realm::Realm::make_shared_realm(realm::Realm::Config, std::__1::shared_ptr<realm::_impl::RealmCoordinator>)::make_shared_enabler::make_shared_enabler(realm::Realm::Config, std::__1::shared_ptr<realm::_impl::RealmCoordinator>) + 80
15 Realm                          0x103f3d428 std::__1::shared_ptr<realm::Realm::make_shared_realm(realm::Realm::Config, std::__1::shared_ptr<realm::_impl::RealmCoordinator>)::make_shared_enabler> std::__1::shared_ptr<realm::Realm::make_shared_realm(realm::Realm::Config, std::__1::shared_ptr<realm::_impl::RealmCoordinator>)::make_shared_enabler>::make_shared<realm::Realm::Config, std::__1::shared_ptr<realm::_impl::RealmCoordinator> >(realm::Realm::Config&&, std::__1::shared_ptr<realm::_impl::RealmCoordinator>&&) + 132
16 Realm                          0x103f37f00 realm::_impl::RealmCoordinator::do_get_realm(realm::Realm::Config, std::__1::shared_ptr<realm::Realm>&, std::__1::unique_lock<std::__1::mutex>&, bool) + 228
17 Realm                          0x103f37d6c realm::_impl::RealmCoordinator::get_realm(realm::Realm::Config) + 296
18 Realm                          0x10400b99c realm::Realm::get_shared_realm(realm::Realm::Config) + 108
19 Realm                          0x103fdafac +[RLMRealm realmWithConfiguration:error:] + 512
20 RealmSwift                     0x1047b9b10 $s10RealmSwift0A0VACyKcfC + 96
21 LimeHDTV                       0x1027a90cc specialized static UseProtocol.migrateRealm() + 122 (FuncOtherStorreClass.swift:122)
22 LimeHDTV                       0x1026a94e8 AppDelegate.application(_:didFinishLaunchingWithOptions:) + 4307358952 (<compiler-generated>:4307358952)
23 LimeHDTV                       0x1026ab04c @objc AppDelegate.application(_:didFinishLaunchingWithOptions:) + 4307365964 (<compiler-generated>:4307365964)
24 UIKitCore                      0x1b27aaa5c -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 356
25 UIKitCore                      0x1b27ac7e8 -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 5048
26 UIKitCore                      0x1b27b2084 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1244
27 UIKitCore                      0x1b1f5946c -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 148
28 UIKitCore                      0x1b2408a38 _UIScenePerformActionsWithLifecycleActionMask + 100
29 UIKitCore                      0x1b1f59f50 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 196
30 UIKitCore                      0x1b1f599a4 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 288
31 UIKitCore                      0x1b1f59d7c -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 740
32 UIKitCore                      0x1b1f5963c -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 336
33 UIKitCore                      0x1b1f5dad8 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke_2 + 188
34 UIKitCore                      0x1b232cdd4 +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:actions:completion:] + 812
35 UIKitCore                      0x1b2421e7c _UISceneSettingsDiffActionPerformChangesWithTransitionContext + 244
36 UIKitCore                      0x1b1f5d834 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke + 140
37 UIKitCore                      0x1b2421d78 _UISceneSettingsDiffActionPerformActionsWithDelayForTransitionContext + 100
38 UIKitCore                      0x1b1f5d69c -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 376
39 UIKitCore                      0x1b1dccf2c __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke + 636
40 UIKitCore                      0x1b1dcbacc -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 248
41 UIKitCore                      0x1b1dccc64 -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 220
42 UIKitCore                      0x1b27b02d0 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 540
43 UIKitCore                      0x1b2352284 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 360
44 FrontBoardServices             0x1b388320c -[FBSSceneImpl _callOutQueue_agent_didCreateWithTransitionContext:completion:] + 424
45 FrontBoardServices             0x1b38a7d84 __86-[FBSWorkspaceScenesClient sceneID:createWithParameters:transitionContext:completion:]_block_invoke.154 + 100
46 FrontBoardServices             0x1b388d7ec -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 232
47 FrontBoardServices             0x1b38a7a40 __86-[FBSWorkspaceScenesClient sceneID:createWithParameters:transitionContext:completion:]_block_invoke + 312
48 libdispatch.dylib              0x1ae3cf524 _dispatch_client_callout + 16
49 libdispatch.dylib              0x1ae3a9808 _dispatch_block_invoke_direct$VARIANT$armv81 + 216
50 FrontBoardServices             0x1b38cc440 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 40
51 FrontBoardServices             0x1b38cc10c -[FBSSerialQueue _queue_performNextIfPossible] + 404
52 FrontBoardServices             0x1b38cc634 -[FBSSerialQueue _performNextFromRunLoopSource] + 28
53 CoreFoundation                 0x1ae687b64 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
54 CoreFoundation                 0x1ae687abc __CFRunLoopDoSource0 + 80
55 CoreFoundation                 0x1ae687244 __CFRunLoopDoSources0 + 184
56 CoreFoundation                 0x1ae682274 __CFRunLoopRun + 788
57 CoreFoundation                 0x1ae681c34 CFRunLoopRunSpecific + 424
58 GraphicsServices               0x1b87cb38c GSEventRunModal + 160
59 UIKitCore                      0x1b27b422c UIApplicationMain + 1932
60 LimeHDTV                       0x10268eacc main + 37 (AppDelegate.swift:37)
61 libdyld.dylib                  0x1ae509800 start + 4
```

## Steps to Reproduce
Does not know
Various attempts to reproduce the error on my own are failed.

## Code Sample
```swift
let config = Realm.Configuration(
    schemaVersion: 12,
    migrationBlock: { migration, oldSchemaVersion in
        if (oldSchemaVersion < 2) {
            migration.enumerateObjects(ofType: SettingApp.className(), { (oldObject, newObject) in
                newObject?["_isFastStart"] = false
                newObject?["_isHideAds"] = false
                newObject?["_titleTextSize"] = 16
                newObject?["_descTextSize"] = 14
                newObject?["_rateTimeoutSec"] = 604800
            })
            migration.enumerateObjects(ofType: SettingMenu.className(), { (old, new) in
                new?["_isShowPanelChannal"] = true
                new?["_typeUseFont"] = 1
            })
            migration.enumerateObjects(ofType: SettingAppPlayList.className(), { (oldObject, newObject) in
                newObject?["listFav"] = List<String>()
                newObject?["listPayPacksID"] = List<String>()
            })
            migration.enumerateObjects(ofType: ProgramModel.className(), { (old, new) in
                new?["_epg"] = ""
            })
            migration.enumerateObjects(ofType: ChannelInfoModel.className(), { (old, new) in
                new?["_statusProgram"] = 0
                new?["_isPublic"] = true
                new?["_href"] = ""
            })
            migration.enumerateObjects(ofType: AdsSettingInterstitialModel.className(), { (old, new) in
                new?["_nexTimeInst"] = 0
            })
            migration.enumerateObjects(ofType: AdsSettingPrerollModel.className(), { (old, new) in
                new?["_nexTimePreRol"] = 0
                new?["_prerollCountEpgInterval"] = 0
                
                new?["_preloadCount"] = 0
                new?["_preloadTimeout"] = 1200
                new?["_preloadBlockTimeout"] = 1200
            })
        }
        else if (oldSchemaVersion < 5) {
            migration.enumerateObjects(ofType: AdsSettingPrerollModel.className(), { (old, new) in
                new?["_preloadCount"] = 2
                new?["_preloadTimeout"] = 1200
                new?["_preloadBlockTimeout"] = 1200
            })
            
            migration.enumerateObjects(ofType: ChannelInfoModel.className(), { (old, new) in
                new?["_href"] = ""
            })
        }
        
        if oldSchemaVersion < 6 {
            migration.enumerateObjects(ofType: InAppPurchaseElementInfoModel.className(), { (old, new) in
                new?["_currency"] = ""
                new?["_isPromo"] = false
                new?["_promoDesc"] = ""
            })
            migration.enumerateObjects(ofType: AdsInfoModel.className(), { (old, new) in
                new?["_name"] = ""
            })
        }
        
        else if oldSchemaVersion < 7 {
            migration.enumerateObjects(ofType: InAppPurchaseElementInfoModel.className(), { (old, new) in
                new?["_isPromo"] = false
                new?["_promoDesc"] = ""
            })
        }
        
        if oldSchemaVersion < 8 {
            migration.enumerateObjects(ofType: ChannelInfoModel.className(), { (old, new) in
                new?["_otherPlayerConfigUrl"] = ""
                new?["_otherPlayerSDK"] = ""
            })
            migration.enumerateObjects(ofType: SettingMenu.className(), { (old, new) in
                new?["_isUseOtherPlayer"] = false
            })
        }
        
        if oldSchemaVersion < 9 {
            migration.enumerateObjects(ofType: SettingApp.className(), { (old, new) in
                new?["_upgrade"] = false
                new?["_lastVersionInAppStore"] = ""
            })
            migration.enumerateObjects(ofType: ChannelInfoModel.className(), { (old, new) in
                new?["_tz"] = 3
            })
            migration.enumerateObjects(ofType: DayModel.className(), { (old, new) in
                new?["isAutoGenerated"] = false
            })
        }
        
        if oldSchemaVersion < 10 { }
        
        if oldSchemaVersion < 11 {
            migration.enumerateObjects(ofType: ProgramModel.className(), { (old, new) in
                new?["channelId"] = old?["_epg"]
            })
        }
        
        if oldSchemaVersion < 12 { }
        
        if oldSchemaVersion < 13 { }
})

Realm.Configuration.defaultConfiguration = config
_ = try? Realm() // CRASH LINE
```

## Version of Realm and Tooling

Realm framework version: 
Realm (4.4.1)
RealmSwift (4.4.1)

Realm Object Server version: ?

Xcode version: 11.5 (11E608c)

iOS/OSX version: Crashlytics data
37% iOS 9
34% iOS 12
14% iOS 13
13% iOS 10

Dependency manager + version: CocoaPods 1.9.1

**Comment by ericjordanmossman** [comment]
@Algashev 
Thanks for the information.

Could you give more context on the circumstance this gets thrown under? Does it occur during all migrations? Or does this occur from non-linear migrations, for example from an old schema version 3 to schema version 12.

A reproduction will help because a potential bug would be otherwise hard to track down.

**Comment by Algashev** [comment]
When I tried to reproduce the error on my own I used linear migration. And my attempts did fail so most probably errors occur from non-linear migrations.

**Comment by jsflax** [comment]
@Algashev , can you try this on the latest version of Realm? Many improvements and fixes have been released since 4.4.1.
